app/Libraries/Payments/CentiliPaymentProcessor.php

```diff
@@ -0,0 +1,130 @@
+<?php
+
+/**
+ *    Copyright 2015-2017 ppy Pty. Ltd.
+ *
+ *    This file is part of osu!web. osu!web is distributed with the hope of
+ *    attracting more community contributions to the core ecosystem of osu!.
+ *
+ *    osu!web is free software: you can redistribute it and/or modify
+ *    it under the terms of the Affero GNU General Public License version 3
+ *    as published by the Free Software Foundation.
+ *
+ *    osu!web is distributed WITHOUT ANY WARRANTY; without even the implied
+ *    warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ *    See the GNU Affero General Public License for more details.
+ *
+ *    You should have received a copy of the GNU Affero General Public License
+ *    along with osu!web.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+namespace App\Libraries\Payments;
+
+use App\Models\Store\Order;
+use Carbon\Carbon;
+use Illuminate\Http\Request;
+
+// FIXME: rename?
+class CentiliPaymentProcessor extends PaymentProcessor
+{
+    private $explodedOrderNumber;
+    private $orderId;
+
+    public function __construct(array $params, PaymentSignature $signature)
+    {
+        parent::__construct($params, $signature);
+        // limiting to 3 means it won't pick up if the format is too long.
```

Comment:
what does this even mean =S

Reply:
`explode('-', '1-2-3-4', 3)` just returns `[1, 2, 3]` so it won't recognise that there's too many elements in the string; doesn't matter since it's going to be replaced with `/^store-(?<userId>\d+)-(?<orderId>\d+)$/`